src/Symfony/Component/Validator/Constraints/HexColorValidator.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+final class HexColorValidator extends ConstraintValidator
+{
+    private const PATTERN_HTML5 = '/^#[0-9a-f]{6}$/i';
+    private const PATTERN = '/^#[0-9a-f]{3}(?:[0-9a-f](?:[0-9a-f]{2}(?:[0-9a-f]{2})?)?)?$/i';
```

Comment:
There is probaby a better pattern? 🤔

Reply:
Maybe I'm missing something, but isn't this equal to `/^[0-9a-f]{3,6}$/i`?